src/engine/controls/cuecontrol.cpp

```diff
@@ -2168,11 +2172,92 @@ void CueControl::updateIndicators() {
             m_pCueIndicator->setBlinkValue(ControlIndicator::ON);
         }
     }
+
+    // Update hotcue indicators: activate the indicator if we are at the cue pos
+    // or if we passed it since the last update, else deactivate.
+    auto prevPos = m_prevPosition.getValue();
+    auto currPos = frameInfo().currentPosition;
+    VERIFY_OR_DEBUG_ASSERT(prevPos.isValid()) {
+        m_prevPosition.setValue(currPos);
+        return;
+    }
+
+    // These are used to check whether we passed the cue while the play position
+    // was wrapped around (loop of full-track repeat).
+    // Note: notifyWrapAround() guarantees that in/out are valid if count > 0,
+    // so we can use count > 0 for loop pos validation.
+    // Possible situations: c=curr, p=prev, |=loop boundaries
+    // count = 0: Regular play:
+    //    P>>>>>>>C
+    // count = 1, no overlap: jump
+    //    |>>>>>C..........P>>>|
+    // count = 1, with overlap (or P=C): fullLoop
+    //    |............P>>>>>>>|
+    //    |>>>>>>>>>>>>C.......|
+    // count > 1: fullLoop
+    //    Multiple wrap-arounds per buffer. May happen with very short loops.
+    //    |............P>>>>>>>|
+    //    |>>>>>>>>>>>>>>>>>>>>|
+    //    |>>>C................|
+    auto loopOut = m_jumpPos;
+    auto loopIn = m_targetPos;
+    bool reverse = false;
+    if (m_wrapAroundCount > 0 && loopOut < loopIn) {
+        // Reverse loop: swap the two so loopIn/Out are always in/out positions.
+        // This reduces the jump/fullLoop comparison cases to one
+        std::swap(loopOut, loopIn);
+        reverse = true;
+    }
+    // We jumped once from out to in (or vice versa).
+    bool jump = m_wrapAroundCount == 1;
+    // That means all loop samples were read and we just check if the cue is
+    // inside the loop.
+    bool fullLoop = m_wrapAroundCount > 1 ||
+            (jump && (prevPos == currPos || reverse == (prevPos < currPos)));
```

Comment:
We have two cases:
```
P>>|>>>>>>>|
    >>>C           
```
reverse = false 
prevPos < currPos = true 

```
|<<<<<<|<<<<P|
   C<<<              
```
reverse = true
prevPos < currPos = false